assets/app/scripts/controllers/overview.js

```diff
@@ -231,14 +233,43 @@ angular.module('openshiftConsole')
       Logger.log("deployments (subscribe)", $scope.deployments);
     }));
 
-    // Sets up subscription for images and imagesByDockerReference
-    watches.push(DataService.watch("images", $scope, function(images) {
-      $scope.images = images.by("metadata.name");
-      $scope.imagesByDockerReference = images.by("dockerImageReference");
-      Logger.log("images (subscribe)", $scope.images);
-      Logger.log("imagesByDockerReference (subscribe)", $scope.imagesByDockerReference);
+    // Sets up subscription for imageStreams
+    watches.push(DataService.watch("imageStreams", $scope, function(imageStreams) {
+      $scope.imageStreams = imageStreams.by("metadata.name");
+      buildDockerRefMapForImageStreams();
+      fetchReferencedImages();
+      // TODO look for "referenced images"
+      Logger.log("imageStreams (subscribe)", $scope.imageStreams);
     }));
 
+    var fetchReferencedImages = function() {
+      angular.forEach($scope.pods, function(pod){
+        angular.forEach(pod.spec.containers, function(container){
+          var dockerRef = container.image;
+          if (!$scope.imagesByDockerReference[dockerRef] && $scope.imageStreamImageRefByDockerReference[dockerRef]) {
+            DataService.get("imageStreamImages", $scope.imageStreamImageRefByDockerReference[container.image], $scope).then(
+              function(image) {
+                $scope.imagesByDockerReference[dockerRef] = image;
+              }
+            );
+          }
+        });
+      })
+    };
+
+    var buildDockerRefMapForImageStreams = function() {
+      angular.forEach($scope.imageStreams, function(imageStream) {
+        angular.forEach(imageStream.status.tags, function(tag){
+          angular.forEach(tag.items, function(image){
+            if (image.image) {
+              $scope.imageStreamImageRefByDockerReference[image.dockerImageReference] = imageStream.metadata.name + "@" + image.image;
```

Comment:
That's the image's "id". Images are not namespaced nor do they belong to image streams (to support deduplication), but only the registry is allowed to retrieve an image solely by its id. Users must retrieve images using imageStreamImages where they specify namespace + image stream + id.

I can't say I ever previously considered having TagEvent.Image be of the form `<namespace>/<image stream>@<id>`. @smarterclayton WDYT?